### PR TITLE
Store and pass abstract working copy types everywhere (except for a few debug commands)

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: cli/src/commands/debug.rs [hunk 1]
```diff
@@ -246,20 +253,43 @@ fn cmd_debug_watchman(
     command: &CommandHelper,
     subcommand: &DebugWatchmanSubcommand,
 ) -> Result<(), CommandError> {
+    use jj_lib::local_working_copy::LockedLocalWorkingCopy;
```
Comment (yuja) [inline cli/src/commands/debug.rs:256]: nit: move `use` to file scope?

File: cli/src/commands/debug.rs [hunk 2]
```diff
@@ -246,20 +253,43 @@ fn cmd_debug_watchman(
...
     match subcommand {
         DebugWatchmanSubcommand::QueryClock => {
-            let (clock, _changed_files) = workspace_command.working_copy().query_watchman()?;
+            let Some(wc): Option<&LocalWorkingCopy> =
+                workspace_command.working_copy().as_any().downcast_ref()
+            else {
+                return Err(user_error(
+                    "This command requires a standard local-disk working copy",
+                ));
+            };
```
Comment (yuja) [inline cli/src/commands/debug.rs:268]: nit: this looks a bit noisy, so how about adding a helper function?

**Review by yuja — APPROVED** [review]

File: cli/src/commands/debug.rs (hunk 1 above)
Comment (martinvonz) [reply cli/src/commands/debug.rs]: Done. VS Code does this sometimes. I haven't figured out when.

File: cli/src/commands/debug.rs (hunk 2 above)
Comment (martinvonz) [reply cli/src/commands/debug.rs]: Done.

**Pull request merged**